src/dicom-parse.c

```diff
@@ -1022,34 +1019,89 @@ char *dcm_parse_frame(DcmError **error,
         .big_endian = is_big_endian(),
     };
 
+    *length = desc->rows * desc->columns * desc->samples_per_pixel * (desc->bits_allocated / 8);
+
+    char *value = DCM_MALLOC(error, *length);
+    if (value == NULL) {
+        return NULL;
+    }
     int64_t position = 0;
+    if (!dcm_require(&state, value, *length, &position)) {
+        free(value);
+        return NULL;
+    }
 
-    if (dcm_is_encapsulated_transfer_syntax(desc->transfer_syntax_uid)) {
-        uint32_t tag;
-        if (!read_tag(&state, &tag, &position) ||
-            !read_uint32(&state, length, &position)) {
+    return value;
+}
+
+/* Read encapsulated frame. Return NULL in case of error.
+*/
+char *dcm_parse_encapsulated_frame(DcmError **error,
+                      DcmIO *io,
```

Comment:
This `DcmIO` should line up with the `DcmError` above. Maybe this is your editor and tabs?

```suggestion
                                   DcmIO *io,
```